### DISCUSSION TITLE
Iwan Kelaiah discussion

### DISCUSSION THREAD

**Discussion by gcarreno** [ObjectPascal-Community/1brc-ObjectPascal#29] [General]
You asked for it:
```console
$ time ./bin/ikelaiah -i /tmp/measurements-1_000_000_000.txt
{...}

real    12m57.737s
user    12m9.575s
sys    0m47.970s
```

**Comment by ikelaiah** [comment]
Thanks for that @gcarreno .
Working for a better algo here. This is fun!

**Comment by gcarreno** [comment]
Hey Iwan(@ikelaiah),

More than welcome!!
Keep at it, please!!

Cheers,
Gus

**Comment by gcarreno** [comment]
Last update:
```console
$ time ./bin/ikelaiah -i /tmp/measurements-1_000_000_000.txt
{...}
real    9m7.152s
user    8m58.213s
sys    0m8.751s
```
This being a cold run.

**Reply by ikelaiah** [reply]
Thank you for posting the results. My laptop runs slower than yours.
Good to see the improvements; from ~15 mins to ~12 mins to ~9mins (on your laptop, mine slower).

**Comment by gcarreno** [comment]
Hey Iwan(@ikelaiah),

Latest `PR` results:
```console
$ ./run_all.sh ikelaiah
******** Run ********

===== Iwan Kelaiah ======
-- SSD --
Benchmark 1: ikelaiah
  Time (mean ± σ):     145.622 s ±  1.904 s    [User: 143.939 s, System: 1.633 s]
  Range (min … max):   142.954 s … 148.707 s    10 runs
 
===========
```
[ikelaiah-1_000_000_000-SSD.json](https://github.com/gcarreno/1brc-ObjectPascal/files/15031407/ikelaiah-1_000_000_000-SSD.json)

Just a disclaimer that this was on a somewhat busy system and you'll have the quiet system results next round.

Cheers,
Gus

**Reply by ikelaiah** [reply]
Thanks for this. The result is very encouraging despite the busy system.

- Iwan

**Reply by gcarreno** [reply]
More than welcome Iwan!!! :people_hugging: :heart:

**Comment by gcarreno** [comment]
Hey Iwan(@ikelaiah),

Finally had time to get all the Free Pascal entries to use `main` aka `trunk`. Even updated to today's commits.

Unfortunately, I was not able to compile `LGenerics`. This means that until I'm able to do so, you're gonna be the only one still in Lazarus 3.0/Free Pascal 3.2.2.

Cheers,
Gus

**Reply by ikelaiah** [reply]
Hi @gcarreno ,

Thanks for letting me know. I'm looking into this now.
Hopefully I can make another PR to rectify this.

-Iwan

**Reply by gcarreno** [reply]
Hey Iwan(@ikelaiah),

You're more than welcome !!

If you can solve the issue quickly, we will get it to `main` aka `trunk` for this run.

I'm only gonna do the run just before I go to sleep and I still have some personal tests to go through.

Cheers,
Gus

**Reply by gcarreno** [reply]
Hey Iwan(@ikelaiah),

I'm really sorry that I didn't catch you `PR` before I ran this week's results.

I've now confirmed that with your last `PR` is compiling under `main` aka `trunk` and the next run will have it so.

Cheers,
Gus

**Reply by ikelaiah** [reply]
Hi @gcarreno ,
Not a problem!

-Iwan
